EditorInput 리팩토링

RBJH opened the pull request:
- 기존 input 태그의 줄바꿈 문제를 해결하기 위해 default 태그 div로 변경
- 앞으로 추가될 이벤트를 위한 조건문 수정
- 한눈에 들어오는 조건문 확인 위해 boolean 변수 선언 후 삼항연산자로 변경

RBJH committed:
Style: EditorInput 컴포넌트 조건문 스타일 변경

기존 한 라인의 조건문을 한눈에 보기 위해 삼항연산자로 변경
기존 리턴 두개의 조건부 렌더링의 코드를 renderTarget 변수로 리턴값 설정

RBJH committed:
Fix: EditorInput 기본 타입 div 태그로 수정

기존 input 태그의 줄바꿈 문제로 다른 태그와 동일하게 div 태그로 변경
다른 key 이벤트를 추가하기 위해 onKeyPress 이벤트 switch문으로 변경

RBJH committed:
Merge: branch 'develop' of https://github.com/connect-foundation/2019-14 into feature/editor

RBJH committed:
Fix: 마크다운 문법 측정 부분 정규표현식으로 수정

앞의 공백 등 마크다운 문법 인식 문제로 인한 정규표현식 사용
공백 4개 시작은 code 태그를 생성해야 하므로 임시로 구현

DrizzlingCattus commented on client/src/components/editor/EditorInput.jsx:
```diff
@@ -52,21 +49,37 @@ const EditorInput = () => {
+      case " ":
+        if (!state.type) setState({ ...state, ...stateAttr[state.value] });
+        break;
+      case "Enter":
```
상수는 따로 빼는게 좋겠네요!

dimsssss reviewed (commented):
고생하셨습니다

DrizzlingCattus commented:
저 ci lint 에러만 해결하면 되겠네요

ImuruKevol commented on client/src/utils/index.js:
```diff
@@ -0,0 +1,15 @@
+  h3: /^ {0,3}### +/gm,
+  h4: /^ {0,3}#### +/gm,
+  h5: /^ {0,3}##### +/gm,
+  h6: /^ {0,3}###### +/gm,
```
혹시 #도 정규식으로 만들 수 있지 않을까요?

RBJH committed:
Feat: 정규 표현식을 통한 마크다운 문법 체크 구현

기존 정규표현식이 제대로 문자열을 인식하지 못하는 부분을 해결하기 위해
멀티라인 옵션 제거 및 기존 white space를 \s로 수정(\t 및 \n 등 고려)

RBJH committed:
Fix: 마크다운 파서 변경에 따른 수정사항 반영

기존 키보드 이벤트를 통해 문법 체크한 부분에 대해 클립보드 복사 시 등
스페이스 바를 누르지 않으면 적용되지 않는 부분을 해결하기 위해 입력 시
정규표현식을 통해 문법에 해당하는 문자열 존재여부 파악하도록 수정
정렬된 리스트의 경우 작성한 숫자부터 start 속성을 주도록 수정

RBJH committed:
Fix: 마크다운 코드 문법 수정

실제 코드 부분 html 태그가 <pre><code></code></pre> 형태이므로
기존 code 만 렌더링하는 부분을 pre로 감싸는 방법으로 수정

RBJH committed:
Feat: 마크다운 수평선 그리기 구현

--- 혹은 *** 시 hr 태그를 구현
- - - 등 지원 문법은 리스트와의 충돌을 막기 위해 지원 제외

RBJH committed:
Fix: 마크다운 문법 및 태그 수정

마크다운 지원 문법 범위가 좁아 확장(___ 등)
hr 태그의 입체감을 없애기 위해 noshade 속성 추가 및 색상 변경
